src/shared/auth.ts

```diff
@@ -16,6 +18,13 @@ export const authOptions: NextAuthOptions = {
   pages: {
     signIn: "/login",
   },
+  callbacks: {
+    async signIn(user) {
+      const loginUser = user.user as AdapterUser;
+      createUser(loginUser);
```

Comment:
```suggestion
      await createUser(loginUser);
```

Reply:
I implemented this change. But why is it needed even when the `createUser` is an async function itself with `await` statements inside it?

Reply:
because if we don't way, we'll redirect before the use is created on the DB, and if you move fast enough to the profile page, you might have a race-condition, where you try to read the profile, but it was not created on the DB yet